T1271

cadecairos opened the pull request:
Updated Google map editor!

Definitely room for improvement still.

Looking for feedback mainly :)

Commit:
1. Rewritten Googlemap editor
2. Added support for Stamen Tiles
3. Live updates to tracks in Butter
4. No terrain or Stamen-Terrain - they are very unstable [#1271 #1257]

Commit:
Added Googlemap editor to default-config.json [#1271]

brettgaylor commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brettgaylor received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/2951028af3a9594/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2951028af3a9594/output.txt

Total script time: 0.75 mins

Server is running on http://pr369.butyr.org/templates/test.html

dseif commented on editors/editor.js:
```diff
@@ -63,7 +63,7 @@ var Comm = function(){
 }; //Comm
 root.Comm = Comm;
-})( window );
+}( window ));
```
why?

cadecairos replied:
JSHint.

dseif commented on editors/googlemap-editor.html:
```diff
@@ -41,420 +26,25 @@
       <br/>
       <label for="lat">&nbsp;Latitude:</label>
-      <input type="text" id="lat" class="small"/>
+      <input type="number" min="-90" max="90" id="lat" class="small"/>
```
is there anyway we could do this with a slider? That would be cool ( if it's supported for the browsers we support )

cadecairos replied:
I don't think sliders would be worthwhile, since Firefox doesn't support that element type (range).

dseif commented on editors/googlemap-editor.html:
```diff
@@ -41,420 +26,25 @@
 
       <label for="lng">&nbsp;Longitude:</label>
-      <input type="text" id="lng" class="small"/>
+      <input type="number" min="-180" max="180" id="lng" class="small"/>
```
same here

cadecairos replied:
same as above.

dseif commented on editors/googlemap-editor.html:
```diff
@@ -41,420 +26,25 @@
+        <!-- DISABLED: UNSTABLE <option value="TERRAIN">Terrain</option>-->
+        <option value="STAMEN-TONER">Stamen Toner</option>
+        <option value="STAMEN-WATERCOLOR">Stamen Water colour</option>
+        <!-- DISABLED: UNSTABLE <option value="STAMEN-TERRAIN">Stamen Terrain</option>-->
```
How are these unstable? They seem to work fine in the default-editor

cadecairos replied:
9 times out of ten they fail to load, and you're left with a gray map div.

humphd replied:
Should we get a bug filed to re-enable and put that bug # in this comment?  Otherwise we'll forget and never come back to this.

cadecairos replied:
I'll file a bug for 0.6

cadecairos replied:
I these types to be a lot better in the latest editor code, so I'm enabling them.

Had to rework how the editor gives new map types in the registry IDs, so that the stamen terrain type didn't overwrite the google one.

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+    trackEvent.location = elements.location.value.trim();
+
+    // zoom
+    trackEvent.zoom = map.getZoom();
```
is there no element for zoom? We should probably have one.

cadecairos replied:
The zoom controls on the map take care of zoom settings.

cadecairos replied:
Do we really need to have a zoom textbox? Also, different maps have different min and max zoom levels, so it's difficult to specify min/max values.

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+    // target
+    trackEvent.target = elements.target.value;
+
+    trackEvent.type = elements.mapType.value;
```
can we change this to be type as well?

cadecairos replied:
I don't follow, what are you asking?

dseif replied:
mapType -> type. Then we could do this in a for loop ( with the exception of zoom as you say they have different zoom values for different maps )

cadecairos replied:
gotcha!

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+
+    _comm.send( "submit", {
+      eventData: trackEvent
+    });
```
Can we iterate over all of the fields instead of doing it line by line? If you name the elements according to the fields we should be able to do that

humphd replied:
What was the conclusion about Dave's comment?

dseif replied:
he put it in a for loop, I guess when you comment on the diff on github the lines move/disappear once changes are made?

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+    select = elements.target;
+    select.innerHTML = "";
+
+    for ( var i = 0, l = targets.length; i < l; i++) {
```
don't know if were going to follow any style guide or not, but whitespace after the `i++`

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+      if (targets[ i ] === trackEvent.target) {
+        select.selectedIndex = i;
+      }
+    }
```
few whitespace issues in this block, if you don't wanna do it that's fine as we have beautifying tickets. Either way, figured I'd make note of it

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+        elements.end.value = trackEvent.end;
+        elements.lat.value = trackEvent.lat || "";
+        elements.lng.value = trackEvent.lng || "";
+        elements.location.value = trackEvent.location || _manifest.options.location[ "default" ];
```
we should make it so we can do this in a loop as well

cadecairos replied:
There's too much going on in this setup to really justify a loop. several of the elements need to be set up, I need to handle defaults in different ways. This only happens once per load anyways.

dseif replied:
what do you mean there's too much going on? looks like we do the same thing as we did with the other loops like I linked you before. For defaulting just  something like:

``` javascript
for ( var item in elements ) {
 elements[ item ].value = trackevent[ item ] || _manifest.options[ item ][ "default" ] || "";
}
```

That should work.

cadecairos replied:
The google map plugin doesn't honour defaults. I have to do them manually.

dseif replied:
do what manually? what do you mean it doesn't honor defaults?

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+
+        // Element Listeners
+        elements.start.addEventListener( "change", update, false );
+        elements.start.addEventListener( "keyup", update, true );
```
why does this need to be true? Is there any other events picking up this listener? Why do we need to change the flow of events?

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+        elements.start.addEventListener( "keyup", update, true );
+
+        elements.end.addEventListener( "change", update, false );
+        elements.end.addEventListener( "keyup", update, true );
```
same here

cadecairos replied:
probably copy pasta cruft.

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,235 @@
+
+        geocode( !!elements.location.value );
+      } else {
+        setTimeout( ready, 15 );
```
doesn't googlemaps give you a way to provide a callback function?

cadecairos replied:
I'm not loading Google maps. it's actually coming from the parent window.

The issue is that dropping a track Event on a template target will open up the editor right away, and in that case, the parent window may not have the google script ready.

dseif replied:
Ah good call, ignore this then

dseif commented on make.js:
```diff
@@ -111,6 +113,10 @@ target['check-lint'] = function() {
+
   exec(JSLINT + ' ' + files + ' --show-non-errors');
 };
 
```
this should be done in another ticket

dseif commented:
Dragging the map around a few times seems to break it. To reproduce, keep dragging to the right when zoomed out it will just go grey and unable to get the map back again.

dseif commented:
Is there a way for us to know if the location string the user is entering is invalid? If I enter some nonsense with numbers and such in it it seems to break and goes grey. We should probably stop this and return to the default if this happens ( or maybe last known good location? ).

dseif commented:
Initial position of the map in the editor is different than that of the one displayed in the template, they should be the same

dseif commented on editors/googlemap-editor.html:
```diff
@@ -4,33 +4,18 @@
 
       <label for="start">&nbsp;From:</label>
-      <input type="text" id="start" class="small"/>
+      <input type="number" min="0" id="start" class="small"/>
```
Also shouldn't be a type number as we allow smpte type strings

Commit:
Fixes and performance enhances for Googlemap Editor [#t1271]

cadecairos commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/9086422771e540b/output.txt

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/e7fe3b8e9fe3dac/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/9086422771e540b/output.txt

Total script time: 0.33 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e7fe3b8e9fe3dac/output.txt

Total script time: 0.87 mins

Server is running on http://pr369.butyr.org/templates/test.html

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,269 @@
+  function update() {
+
+    for ( var prop in _manifest.options ) {
+      if ( _manifest.options.hasOwnProperty( prop ) && !!elements[ prop ] ) {
```
do we need to do this second check? and if so, does it need to be coerced to a  bool to check for existance?

cadecairos replied:
It's good form to filter out unwanted bits of the prototype.

As for the coercion, I don't think it's necessary, but when checking for object existence, it can't hurt.

dseif replied:
fair enough, was just curious why you were doing it.

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,269 @@
+            val = item[ "default" ];
+          } else {
+            val = item.type === "number" ? 0 : "";
+          }
```
Yea as per what humph nailed me on we can do:

``` javascript
if( ..... )
  if ( item[ "default" ] ) {
    return item[ "default" ];
  }
  return item.type === "number" ? 0 : "";
}
return val
```

dseif commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,269 @@
+
+
+        for ( var prop in _manifest.options ) {
+          if ( _manifest.options.hasOwnProperty( prop ) && !!elements[ prop ] ) {
```
same as above. Also if this is similar enough to the top one can we turn it into a function?

cadecairos replied:
I'll look into doing that

dseif replied:
Yea don't worry about this as per our discussion earlier

Commit:
Review fixes to Googlemap Editor [#1271]

dseif commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/bef09e093d7c01b/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/bef09e093d7c01b/output.txt

Total script time: 0.40 mins

humphd commented on editors/googlemap-editor.html:
```diff
@@ -4,33 +4,18 @@
-      }
-
-    </style>
+    <link rel="stylesheet" type="text/css" href="./googlemap-editor.css"></link>
```
Don't bother with the ./ prefix, it's implied.

humphd commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,268 @@
+
+  // Use the geocoder to update the map based on latitude and Longitude
+  // Clears the location textbox
+  function geocode_latLng() {
```
Don't use foo_bar in JS, use fooBar.  Same below.

humphd commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,268 @@
+
+      function defaultValue( item, val ) {
+        if ( val === undefined || typeof val === "object" ) {
+          if ( item[ "default" ] ) {
```
item.default

Commit:
1. Review changes
2. Fix the way stamen types are given ids in the map registry
3. Enabled the terrain types [#1271]

ScottDowne commented on editors/googlemap-editor.js:
```diff
@@ -0,0 +1,268 @@
+
+        // Add Targets
+        addTargets( e.data.targets );
+
```
Extra newline?

ScottDowne commented:
/botio preview

ScottDowne commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d57378806988354/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/305072490cd8f3d/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d57378806988354/output.txt

Total script time: 0.39 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/305072490cd8f3d/output.txt

Total script time: 1.33 mins

ScottDowne commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/2ae51d540fd2a0a/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2ae51d540fd2a0a/output.txt

Total script time: 0.62 mins

Server is running on http://pr369.butyr.org/templates/test.html

Commit:
Removed extra newline [#1271]
